### PR TITLE
add CI via gh-actions

### PR DESCRIPTION
**Pull request by msherman64**

### CODE REVIEW

**Comment by msherman64** [comment]
This is building via QEMU, so is rather slow.
Adding the docker login action should enable "push" to our repo.

Finally, this could be speeded up significantly by using the `--template-only` kolla flag, and building directly using the buildkit cache.

File: zun/kolla-build.conf [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [zun-base]
 type = git
-location = git@github.com:ChameleonCloud/zun.git
-reference = chameleoncloud/train
+location = https://github.com/ChameleonCloud/zun.git
+reference = stable/train
```
Comment (diurnalist) [inline zun/kolla-build.conf:4]:
```suggestion
reference = chameleoncloud/train
```

Needs to be `chameleoncloud/train` to have our patches.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        env:
+          KOLLA_USE_CACHE: no
+          KOLLA_INSTALL_TYPE: source
+          KOLLA_BUILD_PROFILE: zun
+          KOLLA_BASE: ${{ matrix.base_os }}
+          KOLLA_BASE_TAG: ${{ matrix.os_tag }}
+          KOLLA_BASE_ARCH: ${{ matrix.arch }}
+          DOCKER_DEFAULT_PLATFORM: ${{ matrix.platform }}
+        run: |
+          rm -f .env
```
Comment (diurnalist) [inline .github/workflows/build.yml:60]: ?

**Review by diurnalist — APPROVED** [review]
Very cool! We'll have to think about how to trigger builds for the various systems. I'm not sure what is possible there with GHA.

File: .github/workflows/build.yml (hunk 2 above)
Comment (msherman64) [reply .github/workflows/build.yml:60]: You can set it to loop through, e.g. all combinations of [centos8, debian 10] and [x86_65, arm64]

However, in the current kolla-build script, the .env files override existing environment variables.

**Pull request merged**